File: app/src/main/java/com/example/focusstartsecondpart/features/events/data/EventsRepositoryImpl.java
package com.example.focusstartsecondpart.features.events.data;

import com.example.focusstartsecondpart.features.events.domain.EventsRepository;
import com.example.focusstartsecondpart.features.events.domain.model.Event;

import java.util.List;

import io.reactivex.Single;

public class EventsRepositoryImpl implements EventsRepository {

    private EventsDataSource eventsDataSource;

    public EventsRepositoryImpl(EventsDataSource eventsDataSource){
        this.eventsDataSource = eventsDataSource;
    }

    @Override
    public Single<List<Event>> loadEvents() {
        return eventsDataSource.loadEvents();
    }
}

File: app/src/main/java/com/example/focusstartsecondpart/features/events/data/EventsRepositoryImpl.kt
package com.example.focusstartsecondpart.features.events.data

import com.example.focusstartsecondpart.features.events.domain.EventsRepository
import com.example.focusstartsecondpart.features.events.domain.model.Event
import io.reactivex.Single

class EventsRepositoryImpl(private val eventsDataSource: EventsDataSource) : EventsRepository {

    override fun loadEvents(): Single<List<Event>> {
        return eventsDataSource.loadEvents()
    }


}